R/sim_internal.R

```diff
@@ -0,0 +1,204 @@
+#' when `sim_type` is `"contacts"` or `"outbreak"` a contacts table as well.
+#'
+#' @inheritParams sim_linelist
+#' @inheritParams sim_contacts
+#' @inheritParams .check_sim_input
+#'
+#' @return A `<data.frame>` if `sim_type` is `"linelist"` or `"contacts"`, or a
+#' list of two `<data.frame>`s if `sim_type` is `"outbreak"`.
+#' @keywords internal
+.sim_internal <- function(sim_type = c("linelist", "contacts", "outbreak"),
+                          contact_distribution,
+                          contact_interval,
+                          prob_infect,
+                          onset_to_hosp = NULL,
+                          onset_to_death = NULL,
+                          hosp_risk = NULL,
+                          hosp_death_risk = NULL,
+                          non_hosp_death_risk = NULL,
+                          outbreak_start_date,
+                          add_names = NULL,
+                          add_ct = NULL,
+                          min_outbreak_size,
+                          population_age,
+                          case_type_probs = NULL,
+                          contact_tracing_status_probs = NULL,
+                          config) {
+  sim_type <- match.arg(sim_type)
+
+  outbreak_size <- 0
+  max_iter <- 0L
+  # condition on a minimum chain size
+  while (outbreak_size < min_outbreak_size) {
+    .data <- .sim_network_bp(
+      contact_distribution = contact_distribution,
+      contact_interval = contact_interval,
+      prob_infect = prob_infect,
+      config = config
+    )
+    outbreak_size <- sum(.data$infected == "infected")
+    max_iter <- max_iter + 1L
+    if (max_iter >= 1e3) {
+      stop(
+        "Exceeded maximum number of iterations for simulating outbreak. \n",
+        "Change input parameters or min_outbreak_size.",
+        call. = FALSE
+      )
+    }
+  }
+
+  names(.data)[names(.data) == "ancestor"] <- "infector"
+
+  # add delays dates
+  .data$date_onset <- .data$time + outbreak_start_date
+
+  # add exposure date for cases
+  id_time <- data.frame(infector = .data$id, infector_time = .data$time)
+
+  # left join infector time to data preserving column and row order
+  col_order <- c(colnames(.data), "infector_time")
+  .data <- merge(.data, id_time, by = "infector", all.x = TRUE)
+  .data <- .data[order(is.na(.data$infector), decreasing = TRUE), ]
+  .data <- .data[col_order]
+  row.names(.data) <- NULL
+
+  .data <- .add_date_contact(
+    .data = .data,
+    contact_type = "last",
+    distribution = config$last_contact_distribution,
+    config$last_contact_distribution_params,
+    outbreak_start_date = outbreak_start_date
+  )
+  .data <- .add_date_contact(
+    .data = .data,
+    contact_type = "first",
+    distribution = config$first_contact_distribution,
+    config$first_contact_distribution_params
+  )
+
+  # add random age and gender
+  .data$gender <- sample(c("m", "f"), replace = TRUE, size = nrow(.data))
```

Comment:
```suggestion
  # add random age and sex
  .data$sex <- sample(c("m", "f"), replace = TRUE, size = nrow(.data))
```

Gender and sex are not the same constructs, hence, cannot be used interchangeably (even though they are often). I would recommend changing this to "sex" throughout.

Reply:
Thanks for pointing this out. I've had a look through some line list reports and also in the [{outbreaks}](https://www.reconverse.org/outbreaks/) package to check what other data set are doing, and overall it seems to be mixed. I think it's important to make sure this is precise and will make your suggested change throughout the package.

Reply:
Resolved in PR #79.